feat: add GitHub API function, tests and documentation

lskare opened the pull request:
Motivation:
The CI server needs to automate how it creates and updates commit statuses for the commits that are evaluated.

This code connects to GitHubs API function "create commit status" which handles this automatically. Tests are also added to evaluate that all different functionalities of this API endpoint works. Documentation is also added to explain how to set this up on the CI server and locally for GitHub token authentication.

Resolves #3

Commit:
feat: add GitHub API function, tests and documentation

Motivation:
The CI server needs to automate how it creates and updates commit statuses for the commits that are evaluated.

This code connects to GitHubs API function "create commit status" which handles this automatically. Tests are also added to evaluate that all different functionalities of this API endpoint works. Documentation is also added to explain how to set this up on the CI server and locally for GitHub token authentication.

Resolves #3

lskare commented:
If we want to generalise the CI server to handle other repositories then we might want to think about passing the json file from the webhook to the "create_commit_status" function in GitHub_API_functions.py file when it's called from the ci_logic.py file. Atm the owner and repo name for the url which is used in the POST request to GitHubs API is hardcoded to our assignment 2 repo. Simply an idea and not necessary, but good practice, for the assignment

Have an extra look at the documentation and tell me what you think. First time I write documentation.

Stenmarken commented on ci_server/test_GitHub_API_functions.py:
```diff
@@ -0,0 +1,49 @@
+from GitHub_API_functions import create_commit_status
```
The file test_GitHub_API_functions.py should ideally be moved to test_ci_server so that they are run automatically when test_runner is run. This means that you'll also have to adjust the import statement on this line. Something like from ci_server.GitHub_API_functions import create_commit_status should work

Stenmarken reviewed (commented):
Nicely done! Moving the test to test_ci_server might cause annoying path problems so let me know if you need any help!

Commit:
test: Move API function tests to test_ci_server folder

Move the test_GitHubAPI_functions.py file away from the ci_server folder and into the test_server folder in order to have all tests in the same folder
